### PR TITLE
update user guide for coll tuned rules file v2

### PR DESCRIPTION
**Pull request by burlen**
documents #12827

### CODE REVIEW

File: docs/tuning-apps/coll-tuned.rst [hunk 1]
```diff
@@ -128,10 +144,20 @@ for details.
...
 sizes as desired. Simply repeat the sections as needed and adjust the relevant
 count parameters.  One must always provide a rule for message size of zero.
-Message size rules are expected in ascending order.  The last two parameters in
-the rule may or may not be used and have different meaning depending on the
-collective and algorithm.  As of writing not all of the relevant control
-parameters can be set by the rules file (See issue #12589).
+Message size rules are expected in ascending order. The last parameters in the
+message size rule may or may not be used and have different meaning depending
+on the collective and algorithm. The first two parameters in the rule following
+the algorithm id, `topo` and `segment size`, are always required. In version 2
```
Comment (jsquyres) [inline docs/tuning-apps/coll-tuned.rst:150]:
```suggestion
the algorithm ID, `topo` and `segment size`, are always required. In version 2
```

File: docs/tuning-apps/coll-tuned.rst [hunk 2]
```diff
@@ -128,10 +144,20 @@ for details.
...
+of the file format a third parameter, `max requests`, may also be provided. A
+release of OpenMPI at least v5.0.7 or v4.1.8 is required for version 2 features.
```
Comment (jsquyres) [inline docs/tuning-apps/coll-tuned.rst:152]: I know this is a nit, but the official name of the project is "Open MPI" (with a space).  We have to get that correct in the docs.  😄 

```suggestion
release of Open MPI at least v5.0.7 or v4.1.8 is required for version 2 features.
```

File: docs/tuning-apps/coll-tuned.rst [hunk 3]
```diff
@@ -128,10 +144,20 @@ for details.
...
+
+The file format version specifier, `rule-file-version-N` where N is an integer
+greater or equal to 1, should appear on the first line of the file.  If the
+version specifier is not present, the file format is assumed to be version 1.
+Version 2 or greater is required to use the `max requests` parameter. OpenMPI
```
Comment (jsquyres) [inline docs/tuning-apps/coll-tuned.rst:157]:
```suggestion
Version 2 or greater is required to use the `max requests` parameter. Open MPI
```

File: docs/tuning-apps/coll-tuned.rst [hunk 4]
```diff
@@ -128,10 +144,20 @@ for details.
...
+releases older than v5.0.7 and v4.1.8 do not support the file format version
```
Comment (jsquyres) [inline docs/tuning-apps/coll-tuned.rst:158]: Open MPI v4.1.8 does not exist, and may never exist (unless there's a strong need, v4.1.7 may well be the last v4.1.x version released).

File: docs/tuning-apps/coll-tuned.rst [hunk 5]
```diff
@@ -128,10 +144,20 @@ for details.
...
+identifier. When using older releases of OpenMPI do not include a version
```
Comment (jsquyres) [inline docs/tuning-apps/coll-tuned.rst:159]:
```suggestion
identifier. When using older releases of Open MPI do not include a version
```

**Review by jsquyres — CHANGES_REQUESTED** [review]

File: docs/tuning-apps/coll-tuned.rst (hunk 1 above)
Comment (burlen) [reply docs/tuning-apps/coll-tuned.rst]: fixed

File: docs/tuning-apps/coll-tuned.rst (hunk 2 above)
Comment (burlen) [reply docs/tuning-apps/coll-tuned.rst]: fixed

File: docs/tuning-apps/coll-tuned.rst (hunk 3 above)
Comment (burlen) [reply docs/tuning-apps/coll-tuned.rst]: fixed

File: docs/tuning-apps/coll-tuned.rst (hunk 5 above)
Comment (burlen) [reply docs/tuning-apps/coll-tuned.rst]: fixed

**Comment by burlen** [comment]
I fixed those issues

**Comment by jsquyres** [comment]
@burlen Thank you!  Can you cherry pick over to v5.0.x?

**Pull request merged**